discovery/gossiper_test.go

```diff
@@ -381,7 +452,11 @@ func (r *mockGraphSource) IsStaleEdgePolicy(chanID lnwire.ShortChannelID,
 // NOTE: This method is part of the ChannelGraphSource interface.
 func (r *mockGraphSource) MarkEdgeLive(chanID lnwire.ShortChannelID) error {
 	r.mu.Lock()
-	defer r.mu.Unlock()
+	defer func() {
+		r.callCount["MarkEdgeLive"]++
```

Comment:
An alternative way to accomplish this would be to use `mock.Mock`. Then you can just do `mock.AssertNumberOfCalls(t, "MarkEdgeLive", 3)`. You can also set the expectations upfront as well.

Reply:
https://pkg.go.dev/github.com/stretchr/testify/mock#Mock.AssertNumberOfCalls

Reply:
yeah it would be great if we can gradually replace all mockers using `mock.Mock`, tho that would involve quite some refactor of the tests here. So maybe a PR at the end of the graph cleanup series?

Reply:
> So maybe a PR at the end of the graph cleanup series?

Yeah sounds like a plan 🫡